### ISSUE TITLE
array-bracket-newline doesn't count multiline comment as multiline

### ISSUE THREAD

**Issue by TWiStErRob** [eslint/eslint#9211]
**Tell us about your environment**

* **ESLint Version:** 4.6.0
* **Node Version:** 5.6.0
* **npm Version:** 3.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
{
	"rules": {
		"array-bracket-newline": ["warn", {
			"multiline": true
		}]
	}
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
var structure = {
	marriages: [/*
		{
			husband: { id: 1, name: "Hubby" },
			wife: { id: 2, name: "Wifey" }
		},
		...
	*/],
	// more stuff here
}
```

**What did you expect to happen?**
no warnings

**What actually happened? Please include the actual, raw output from ESLint.**
```
  2:13  warning  There should be no linebreak after '['   array-bracket-newline
  8:4   warning  There should be no linebreak before ']'  array-bracket-newline
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I can reproduce this issue.

**Comment by i-ron-y** [comment]
What is the intended behaviour here?

Should the original example be considered practically the same as ...

    var structure = {
	    marriages: [/* */],
	    // more stuff here
    }

... ? (i.e. The array in the original example technically doesn't have elements on a new line because all the new lines are part of the comment, which isn't an element.)

**Comment by not-an-aardvark** [comment]
Based on the [documentation](https://eslint.org/docs/rules/array-bracket-newline#options), it seems like newlines should only be required in the brackets "if there are line breaks inside elements or between elements". Since the array in the example doesn't have any elements, there aren't any line breaks inside elements or between elements. So I think linkebreaks should be disallowed in this example, which means that the original code should be considered correct because it doesn't have linebreaks after the `[` or before the `]`.

**Comment by platinumazure** [comment]
This might be another case where it could make sense to work based on token/comment line breaks instead of element line breaks.

**Comment by philquinn** [comment]
I have a PR to address this. Any reviews are welcome.

**Comment by TWiStErRob** [comment]
Awesome, thank you! Confirmed working on my codebase.
